api/v1/ptpconfig_webhook.go

```diff
@@ -194,6 +226,230 @@ func (r *PtpConfig) validate() error {
+				continue
+			}
+			if profile.Ptp4lConf == nil {
+				continue
+			}
+
+			conf := &ptp4lConf{}
+			if err := conf.populatePtp4lConf(profile.Ptp4lConf, profile.Ptp4lOpts); err != nil {
+				continue
+			}
+
+			if globalSection, exists := conf.sections["[global]"]; exists {
+				if saFile, exists := globalSection.options["sa_file"]; exists && saFile != "" {
+					// Check if THIS PtpConfig uses the same sa_file
+					if currentSecret, found := currentSaFileToSecret[saFile]; found {
+						// Conflict: same sa_file but different secret
+						if currentSecret != *profile.PtpSecretName {
+							return fmt.Errorf("sa_file '%s' conflict: PtpConfig '%s' already uses secret '%s' with this sa_file path, but this PtpConfig tries to use secret '%s'. All PtpConfigs using the same sa_file must reference the same secret",
+								saFile, existingConfig.Name, *profile.PtpSecretName, currentSecret)
+						}
+					}
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+// validateSecretExists checks that ptpSecretName references an existing secret
+func (r *PtpConfig) validateSecretExists(ctx context.Context) error {
+	if webhookClient == nil {
+		ptpconfiglog.Info("webhook client not initialized, skipping secret existence validation")
+		return nil
+	}
+
+	for _, profile := range r.Spec.Profile {
+		// Skip if no secret specified
+		if profile.PtpSecretName == nil || *profile.PtpSecretName == "" {
+			continue
+		}
+
+		secretName := *profile.PtpSecretName
+		profileName := "unknown"
+		if profile.Name != nil {
+			profileName = *profile.Name
+		}
+
+		// Try to get the secret from openshift-ptp namespace
+		secret := &corev1.Secret{}
+		err := webhookClient.Get(ctx, types.NamespacedName{
+			Namespace: "openshift-ptp",
+			Name:      secretName,
+		}, secret)
+
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return fmt.Errorf("secret '%s' referenced by profile '%s' does not exist in namespace 'openshift-ptp'. Please create the secret before referencing it in PtpConfig",
+					secretName, profileName)
+			}
+			// For other errors (like permission issues), log but don't block
+			ptpconfiglog.Error(err, "failed to verify secret existence", "secret", secretName, "profile", profileName)
+			// Fail open - don't block if we can't verify
+			return nil
+		}
+
+		ptpconfiglog.Info("validated secret exists", "secret", secretName, "profile", profileName)
+
+		// Validate SPP (Security Parameter Profile) if specified
+		if err := r.validateSppInSecret(profile, secret); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// validateSppInSecret checks that the spp value in ptp4lConf exists in the referenced secret
+func (r *PtpConfig) validateSppInSecret(profile PtpProfile, secret *corev1.Secret) error {
+	// Skip if no ptp4lConf
+	if profile.Ptp4lConf == nil {
+		return nil
+	}
+
+	// Parse ptp4lConf to get spp value from [global] section
+	conf := &ptp4lConf{}
+	if err := conf.populatePtp4lConf(profile.Ptp4lConf, profile.Ptp4lOpts); err != nil {
```

Comment:
The r.validate() func is already populating the ptp4lConf when it iterates over the profiles. Isn't it possible to reuse [that one](https://github.com/k8snetworkplumbingwg/ptp-operator/pull/123/files#diff-49d9e80697fdeec0d139aa476343d0fbb1a04127dc13bb69654138b33021f6d4R140)?